.github/workflows/changelog.yml

```diff
@@ -6,19 +6,19 @@ on:
     types: [opened, reopened, labeled, unlabeled, synchronize]
 
 env:
-  FRAGMENT_NAME: "docs/changes/${{ github.event.number }}.{feature,bugfix,api,datamodel,optimization,maintenance}.rst"
+  FRAGMENT_NAME_PREFIX: "docs/changes/${{ github.event.number }}"
 
 jobs:
   changelog:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout code
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Check for news fragment
-        if: ${{ ! contains( github.event.pull_request.labels.*.name, 'no-changelog-needed')}}
-        uses: andstor/file-existence-action@v3
-        with:
-          files: ${{ env.FRAGMENT_NAME }}
-          fail: true
+      - name: Check for at least one changelog fragment
+        if: ${{ !contains(github.event.pull_request.labels.*.name, 'no-changelog-needed') }}
```

Comment:
Are you absolutely sure that no space after "!" works as you want? The evaluation rules for bash are very fickle and specific.

Reply:
yes. This PR has the "no-changelog-needed" label and passes the changelog check despite it has no changelog entry.

Reply:
This is not bash, it is githubs expression language